### PR TITLE
Add ibc3 and FairburnPool

### PR DESCRIPTION
**Pull request by jhernandezb**

### CODE REVIEW

File: app/genesis.go [hunk 1]
```diff
@@ -30,7 +30,6 @@ func NewDefaultGenesisState(cdc codec.JSONCodec) GenesisState {
 		Params: wasmtypes.Params{
 			CodeUploadAccess:             wasmtypes.AllowNobody,
 			InstantiateDefaultPermission: wasmtypes.AccessTypeEverybody,
-			MaxWasmCodeSize:              DefaultMaxWasmCodeSize,
```
Comment (shanev) [inline app/genesis.go:33]: Where is this defined now?

File: cmd/starsd/cmd/genesis.go [hunk 2]
```diff
@@ -468,7 +468,7 @@ func MainnetGenesisParams() GenesisParams {
 	genParams.ConsensusParams.Version.AppVersion = 1
 
 	genParams.WasmParams = wasmtypes.DefaultParams()
-	genParams.WasmParams.MaxWasmCodeSize = 1000 * 1024 * 2 // 2000 KB
+	// genParams.WasmParams.MaxWasmCodeSize = 1000 * 1024 * 2 // 2000 KB
```
Comment (shanev) [inline cmd/starsd/cmd/genesis.go:471]: Did you mean to comment this out?

File: cmd/starsd/cmd/genesis.go [hunk 3]
```diff
@@ -510,7 +510,7 @@ func TestnetGenesisParams() GenesisParams {
 	}
 	genParams.WasmParams.CodeUploadAccess = wasmtypes.AllowEverybody
 	genParams.WasmParams.InstantiateDefaultPermission = wasmtypes.AccessTypeEverybody
-	genParams.WasmParams.MaxWasmCodeSize = 1000 * 1024 * 2 // 1000kb
+	// genParams.WasmParams.MaxWasmCodeSize = 1000 * 1024 * 2 // 1000kb
```
Comment (shanev) [inline cmd/starsd/cmd/genesis.go:513]: Same

File: cmd/starsd/cmd/genesis.go [hunk 4]
```diff
@@ -554,6 +554,6 @@ func LocalnetGenesisParams() GenesisParams {
 	genParams.GovParams.TallyParams.Quorum = sdk.MustNewDecFromStr("0.1")    // 10%
 	genParams.GovParams.TallyParams.Threshold = sdk.MustNewDecFromStr("0.5") // 50%
 	genParams.GovParams.VotingParams.VotingPeriod = time.Minute * 1          // 5 min
-	genParams.WasmParams.MaxWasmCodeSize = 1000 * 1024 * 2                   // 2000kb
+	// genParams.WasmParams.MaxWasmCodeSize = 1000 * 1024 * 2                   // 2000kb
```
Comment (shanev) [inline cmd/starsd/cmd/genesis.go:557]: Same

File: x/alloc/keeper/msg_server_fund_fee_collector.go [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+	sender, err := sdk.AccAddressFromBech32(msg.Sender)
+	if err != nil {
+		return nil, err
+	}
+	err = k.FundFairburlPool(ctx, sender, msg.Amount)
+	if err != nil {
+		return nil, err
+	}
+	ctx.EventManager().EmitEvent(
```
Comment (shanev) [inline x/alloc/keeper/msg_server_fund_fee_collector.go:21]: There's also a Fair Burn event on the contract side. Maybe we should get rid of that one and keep this. Then there'll no longer be a need to pass in a mutable `Response`.

File: app/genesis.go (hunk 1 above)
Comment (jhernandezb) [reply app/genesis.go:33]: this was removed in wasmd 0.27

File: app/app.go [hunk 6]
```diff
@@ -425,14 +427,16 @@ func NewStargazeApp(
 	// Create Transfer Keepers
 	app.TransferKeeper = ibctransferkeeper.NewKeeper(
 		appCodec, keys[ibctransfertypes.StoreKey], app.GetSubspace(ibctransfertypes.ModuleName),
-		app.IBCKeeper.ChannelKeeper, &app.IBCKeeper.PortKeeper,
+		app.IBCKeeper.ChannelKeeper, app.IBCKeeper.ChannelKeeper, &app.IBCKeeper.PortKeeper,
```
[the remark this answers is not in the document]
Comment (jhernandezb) [reply app/app.go:430]: It requires a ChannelKeeper interface and a PortKeeper interface (new in ibc3) but in this case IBCKeeper fulfills both

File: cmd/starsd/cmd/genesis.go (hunk 2 above)
Comment (jhernandezb) [reply cmd/starsd/cmd/genesis.go:471]: needs to be removed

File: cmd/starsd/cmd/genesis.go (hunk 3 above)
Comment (jhernandezb) [reply cmd/starsd/cmd/genesis.go:513]: removed

File: x/alloc/keeper/msg_server_fund_fee_collector.go (hunk 5 above)
Comment (jhernandezb) [reply x/alloc/keeper/msg_server_fund_fee_collector.go:21]: this only does the funding part and the burn side still part of the contracts we could probably do both things here like a single message that does burn and fund.

Comment (shanev) [reply x/alloc/keeper/msg_server_fund_fee_collector.go:21]: Oh that's right. Either way is fine. Since this already works can just leave it as is.

**Review by shanev — APPROVED** [review]

**Pull request merged**